### ISSUE TITLE
De-inline Javascript

### ISSUE THREAD

**Issue by colinmollenhour** [hps/heartland-magento-extension#56]
I figured out that the script tag does not load when the checkout form is loaded via Ajax. This is the problem that people were having with the checkout-form.js file. To fix this you can use:

```php
<?php /* Load script using this method in case first load is via Ajax */ ?>
<script type="text/javascript">
  (function(){
      var script = document.createElement('script');
      script.type = 'text/javascript';
      script.src = '<?php echo $this->getJsUrl('securesubmit/checkout-form.js').'?m='.filemtime(Mage::getBaseDir().DS.'js'.DS.'securesubmit'.DS.'checkout-form.js') ?>';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
  })();
</script>
```

You've already inlined this checkout-form.js file anyway, but if you want to go back to not inlining it this should fix all of the problems associated with loading from an external file.

**Comment by jdrewlayton** [comment]
Where would you suggest putting this?

**Comment by colinmollenhour** [comment]
I am talking about removing the huge block of Javascript from the .phtml file and putting it back in a .js file and loading it using a script tag. Does that answer your question?

**Comment by slogsdon** [comment]
We've had a lot of the Javascript out of `form.phtml` for a while now (excluding the gift card code that is currently there), but we managed to forget to close this issue. We are currently in the process of revamping the user experience and are starting work on cleaning up the Javascript even more, creating a [simple entry point](https://github.com/hps/heartland-magento-extension/blob/magental/app/design/frontend/base/default/template/securesubmit/form.phtml#L122-L142) in the process.

@colinmollenhour Feel free to open this back up if you think it could be de-inlined more.

**Comment by colinmollenhour** [comment]
Ahh yes looks like you have made a lot of changes since this ticket was opened. Definitely doesn't need any more inlining, although the method I propose has one advantage: it works for people that already loaded the checkout page before SecureSubmit was enabled (since it is loaded when the payment methods block is loaded). This is an edge case but occasionally it is necessary to switch payment methods on and off and this would cause interruption of checkout to fewer customers during transition.
